app/models/assignment.rb

```diff
@@ -1156,6 +1156,31 @@ def to_json(options = {})
     self.assignment_properties.attributes.merge(self.attributes).symbolize_keys.to_json(options)
   end
 
+  # Returns an assignment's relevant properties for uploading/downloading an assignment's configuration as a hash
+  def assignment_properties_config
+    properties = {}
+    # Insert DEFAULT_FIELDS first to make file more familiar
+    DEFAULT_FIELDS.each do |f|
+      properties[f] = self.assignment.send(f)
+    end
+    # Get all other assignment properties
+    other_properties = self.assignment_properties.attributes.merge(self.attributes).symbolize_keys
+    section_dates = SectionDueDate.where(assessment_id: self.id).pluck_to_hash(:due_date, :section_id, :start_time)
```

Comment:
use `self.section_due_dates` instead of `SectionDueDate.where(assessment_id: self.id)`